Samples/Samples/ViewModel/CalendarEventAddViewModel.cs

```diff
@@ -322,22 +376,74 @@ public bool IsMonthDaySpecific
                     {
                         SelectedRecurrenceMonthDay = 0;
                         SelectedRecurrenceMonthWeek = IterationOffset.First;
-                        SelectedMonthWeekRecurrenceDay = DayOfTheWeek.Monday;
+                        SelectedMonthWeekRecurrenceDay = CalendarDayOfWeek.Monday;
                     }
                 }
             }
         }
 
-        public ObservableCollection<DayOfTheWeekSwitch> RecurrenceDays { get; set; } = new ObservableCollection<DayOfTheWeekSwitch>()
+        bool IsUpdatingCheckBoxGroup { get; set; } = false;
+
+        void SetCheckBoxes(int bitFlagValue)
         {
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Monday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Tuesday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Wednesday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Thursday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Friday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Saturday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Sunday }
-        };
+            try
+            {
+                IsUpdatingCheckBoxGroup = true;
+                var currentVal = bitFlagValue;
+                var maxValue = (int)CalendarDayOfWeek.Saturday;
+                for (var i = maxValue; i > 0; i /= 2)
+                {
+                    RecurrenceDays[(int)Math.Log(i, 2)].IsChecked = currentVal >= i;
+
+                    if (currentVal >= i)
+                    {
+                        currentVal -= i;
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
+            finally
+            {
+                IsUpdatingCheckBoxGroup = false;
+            }
+        }
+
+        void OnChildCheckBoxChangedEvent(object sender, PropertyChangedEventArgs e)
+        {
+            if (sender == null)
+            {
+                return;
+            }
+            if (!(sender is CalendarDayOfWeekSwitch calendarDayOfWeek))
+            {
+                return;
+            }
+            if (!IsUpdatingCheckBoxGroup)
+            {
+                OnPropertyChanged(nameof(IsNone));
+                OnPropertyChanged(nameof(IsWeekdays));
+                OnPropertyChanged(nameof(IsWeekend));
+                OnPropertyChanged(nameof(IsAllDays));
+            }
+        }
+
+        ObservableCollection<CalendarDayOfWeekSwitch> recurrenceDays;
```

Comment:
Is this in the correct order in the file? 
Should this be readonly now?

Reply:
I'm not sure if it can be readonly, as if I try to make it readonly it wants OnChildCheckBoxChangedEvent to be static, which doesn't let me call OnPropertyChanged

Reply:
Hmmm that shouldn't be the case - you're assuming that a readonly field has to be set inline - readonly field can be set in the constructor like you are doing now I think. If not, change to readonly property

Reply:
I see, yeah funny that, when I'm trying to understand all this complex features of MVVM I'm forgetting the basics of a constructor can be used to set readonly fields